__tests__/pollySetup.ts

```diff
@@ -18,6 +29,31 @@ export const setupPollyMiddleware = (server: PollyServer) => {
     recording.request.headers = requestHeaders;
     recording.response.cookies = [];
     recording.response.headers = responseHeaders;
+
+    if (recording.request.postData.text.includes("FetchAppMetafields")) {
+      const responseData = JSON.parse(recording.response.content.text);
```

Comment:
`content.text` can be undefined and might not be in a JSON format. I had this case in checkout app tests (especially error responses). You could add a fallback for this:

```ts
const tryParse = (data: string | undefined) => {
  try {
    return JSON.prase(data)
  } catch (e) {
    return data;
```

since content can be empty, you should also do this:

```suggestion
      const responseData = tryParse(recording.response.content?.text);
```